### PR TITLE
add example for flash attnetion

### PR DESCRIPTION
**Pull request by JackCaoG**
This is based on https://github.com/huggingface/transformers/compare/main...pytorch-tpu:transformers:alanwaketan/flash_attention

### CODE REVIEW

**Review by alanwaketan — APPROVED** [review]

File: examples/fsdp/train_decoder_only_fsdp_v2.py [hunk 1]
```diff
@@ -26,7 +26,8 @@ def __init__(self):
     mesh_shape = (num_devices, 1)
     device_ids = np.array(range(num_devices))
     # To be noted, the mesh must have an axis named 'fsdp', which the weights and activations will be sharded on.
-    mesh = xs.Mesh(device_ids, mesh_shape, ('fsdp', 'model'))
+    self.mesh = xs.Mesh(device_ids, mesh_shape, ('fsdp', 'model'))
```
[the remark this answers is not in the document]
Comment (JackCaoG) [reply examples/fsdp/train_decoder_only_fsdp_v2.py:29]: yea I used it below. I guess I can remove it from the `self`.

We still need to explicitly pass `mesh` to dataloader through right?

[the remark this answers is not in the document]
Comment (alanwaketan) [reply examples/fsdp/train_decoder_only_fsdp_v2.py:29]: dataloader is a bit different... it accepts ShardingSpec instead of mesh and partition_spec....

**Comment by vanbasten23** [comment]
Do you think we should add the commands to run the examples, especially when they need some flags?

**Comment by JackCaoG** [comment]
> Do you think we should add the commands to run the examples, especially when they need some flags?

you don't need any flags beside `PJRT_DEVICE`, you can just run it with `python train_xxxx.py`

**Pull request merged**